src/Core/Content/ProductExport/Service/ProductExportGenerator.php

```diff
@@ -196,15 +176,46 @@ public function generate(ProductExportEntity $productExport, ExportBehavior $exp
                     continue; // Skip variants unless they are included
                 }
 
-                $body .= $this->productExportRender->renderBody($productExport, $context, $data);
+                $total++;
+                
+                if (!$batchItterationBreak) {
+                    $body .= $this->productExportRender->renderBody($productExport, $context, $data);
+                }
             }
-
+            
             if ($exportBehavior->batchMode()) {
-                break;
+                $batchItterationBreak = true;
             }
         }
+        //Throw Error when no products are to be exported
+        if ($total === 0) {
```

Comment:
Can you add testcases for the changed behaviour?

Reply:
So you need a test for the new total count?
The thrown error is from the original file, just moved downwards after the accurate total count.

Reply:
yep every bugfix should go with a testcase that failed before the fix, that is the only way we can ensure that the bug does not pop up again